pkg/gcp/gcp_lock.go

```diff
@@ -20,10 +21,14 @@ func (googleLock *GoogleStorageLock) Lock(transactionId int, resource string) (b
 	if err != nil {
 		fmt.Printf("failed to get lock: %v\n", err)
 	}
-	if existingLockTransactionId != nil {
+	if existingLockTransactionId != nil && *existingLockTransactionId != transactionId {
```

Comment:
I think this kind logic should actually leave a layer above, otherwise we will have to repeat it for each lock provider

Reply:
Right, fixed